src/rules/strictComparisonsRule.ts

```diff
@@ -120,9 +120,68 @@ export class Rule extends Lint.Rules.TypedRule {
     }
 }
 
+function getConstraint(t: ts.Type): ts.TypeReferenceNode | undefined {
+    if (t.getSymbol() === undefined) {
+        return undefined;
+    }
+
+    const paramDeclaration = t.symbol.declarations.find(
+        d => d.kind === ts.SyntaxKind.TypeParameter,
```

Comment:
Yeah that `as` is a little unfortunate... `tsutils'` methods annotated with TypeScript's `is` return syntax should help with that:

```suggestion
        tsutils.isTypeParameter,
```

If strict equality checking against `undefined` gets in your way, you could either leave this as is, or use `filter` instead and check for `.length` I suppose...